src/sieve.py

```diff
@@ -1,20 +1,38 @@
-"""Computing primes."""
-
-
-def sieve(n: int) -> list[int]:
-    """
-    Compute all the primes up to (and possibly including) `n`.
-
-    `n` must be positive.
-
-    >>> sieve(15)
-    [2, 3, 5, 7, 11, 13]
-
-    """
-    assert n > 0
-    candidates = list(range(2, n + 1))
-    primes = []
-
-    # FIXME: fill out this bit
-
-    return primes
+"""Computing primes."""
+
+
+def sieve(n: int) -> list[int]:
+    """
+    Compute all the primes up to (and possibly including) `n`.
+
+    `n` must be positive.
+
+    >>> sieve(15)
+    [2, 3, 5, 7, 11, 13]
+
+    """
+    assert n > 0
+    candidates = list(range(2, n + 1))
+    primes = []
+
+    # FIXME: fill out this bit
+
+    # first candidate is prime
+    prime = candidates.pop(0)

```

Comment:
I wouldn't recommend using `pop(0)` to get the first element. It is a slow operation, as it will copy all the remaining elements of `candidates` one step up. If is fast to pop the last element of a list but slow to pop the first. We haven't talked about efficiency yet, so that might not make sense this week, but maybe it will next week.

It is generally better to just use indices, or to reverse the list and pop from the other end.